### ISSUE TITLE
Implement the attack logic

### ISSUE THREAD

**Issue by LucasLvy** [keep-starknet-strange/tsubasa#26]
# Description

As per the game rules, at the end of your turn, all your cards on the field will try to pass the opponents' players. If you dribble an enemy player it'll go out of the game but will reduce your defense by his dribble stats. If the opponent's board is empty you'll score a goald and win the round. The attacking sequence is : attacker => midfielder => defense => goalkeeper
(this should be in the end turn system)

# Acceptance criteria

* The attacking sequence is respected
* Cards that get passed get removed from the board
* If a card doesn't get passed but gets attacked its defense stat is updated

**Comment by dbejarano820** [comment]
i can take this

**Comment by LucasLvy** [comment]
Hey any news there N

**Comment by LucasLvy** [comment]
It is blocking other PRs so unassigning it. Feel free to pick up something else

**Comment by WhoIsNac** [comment]
Sorry, I didn't see the message, I've started the attack logic and contract and implemented the test, (I've also created a card creation contract to test with card stats). 

But to finish the implementation, we need to finish the card logic system and also the place_card system to know every card in the field in order to start the attack sequence between the players' cards. 

I don't know if we're allowed to implement other parts to finish this one.
